lib/smb2/file.rb

```diff
@@ -119,4 +165,51 @@ def size
     info.end_of_file
   end
 
+  # Write the entire contents of `data`, starting at `offset` bytes from the
+  # beginning of the file.
+  #
+  # @param data [String] what to write
+  # @param offset [Fixnum] where in the file to start writing
+  # @return [Fixnum] number of bytes written. This may be less than the length
+  #   of `data` if there was an error.
+  def write(data, offset: self.pos)
+    max = tree.client.max_write_size
+    bytes_written = 0
+    seek(offset)
+
+    while data.length > bytes_written
+      data_chunk = data.slice(bytes_written, max)
+      response_packet = write_chunk(data_chunk, offset: pos)
+
+      # @todo raise instead?
+      break if response_packet.header.nt_status != 0
```

Comment:
You could do `break unless response.header.nt_status == 0` since you're using `!=`.